src/spec/repeater-task-queue.spec.coffee

```diff
@@ -0,0 +1,75 @@
+SphereClient = require '../lib/client'
+RepeaterTaskQueue = require '../lib/repeater-task-queue'
+Config = require('../config').config
+util = require('util')
+describe 'RepeaterTaskQueue', ->
+
+  repeaterOptions = {
+    attempts: 5,
+    timeout: 200
+  }
+
+  beforeEach ->
+    task = new RepeaterTaskQueue { maxParallel: 30 }, { attempts: 5, timeout: 200, timeoutType: 'v' }
+    sphereConfig = { config: Config, task }
+    @client = new SphereClient sphereConfig
+
+
+  it 'should finally resolve after three tries', (done) ->
+
+    callsMap = {
+      0: { statusCode: 500, message: 'ETIMEDOUT' }
+      1: { statusCode: 500, message: 'ETIMEDOUT' }
+      2: { statusCode: 200, message: 'success' }
+    }
+    callCount = 0
+    spyOn(@client._rest, 'GET').andCallFake (resource, callback) ->
```

Comment:
It's a devDependency and jasmine is not know for stubbing. The cons are not much but I have no strong opinion on this, so fine with me this way.